### PR TITLE
ut: add cron-fhpa validation unit-test

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add cron-fhpa validation unit-test

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/webhook/cronfederatedhpa/validating_test.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+					Name:       "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
```
Comment (jwcesign) [inline pkg/webhook/cronfederatedhpa/validating_test.go:35]: We can add the normal case of scaling Deployment and FederatedHPA.

File: pkg/webhook/cronfederatedhpa/validating_test.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+				},
+			},
+			expectedErr: "may not be '..'",
+		},
+		{
+			name: "invalid spec.scaleTargetRef.name: contains %",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: "foo",
+					Name: "1%100",
```
Comment (jwcesign) [inline pkg/webhook/cronfederatedhpa/validating_test.go:147]: Do we need to test the scaleTargetRef in such detail? Will it make the unit test code difficult to maintain? Essentially, this is a function call to K8s.

**Review by jwcesign — COMMENTED** [review]
Nice work, thanks! Other lgtm.

**Comment by whitewindmills** [comment]
PTAL @jwcesign

**Review by jwcesign — COMMENTED** [review]
/lgtm

/cc @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**